sdk/java/pulumi/src/main/java/com/pulumi/deployment/internal/DeploymentTests.java

```diff
@@ -341,4 +383,27 @@ public static Log mockLog(Logger logger) {
     public static Log mockLog(Logger logger, Supplier<Engine> engine) {
         return new Log(new DefaultEngineLogger(logger, () -> Mockito.mock(Runner.class), engine));
     }
+
+    private static final class OutputsCaptureHelper {
```

Comment:
what is this monstrosity?

Reply:
It's an internal detail for keeping track of the `Map<String, Output<?>>` stack outputs coming out of the user-provided code, so that they can be exposed in the Result class. It's much easier to have it in a little class than inlining this into the call site but YMMV.

Reply:
this looks like accidental complexity hiding bad architecture, there are better ways

Reply:
Looks like the method that this supports has 1 call site. Let me see if this can just be deleted.

Reply:
That worked. This is gone.